### PR TITLE
ci: update flags and descs for E2E script

### PR DESCRIPTION
**Pull request by nathan-weinberg**
This PR changes the following:
- Changes `f` flag to `L` and renames `FULLSIZE` training to `LEGACYTRAIN` to more accurately reflect what the flag is doing
- Updates ordering of `basic-workflow-tests.sh` flags to be alphabetical in both script `usage` func, `getopts` func, and docs
- Updates desc of several of the flags to be more in-depth
- Adds phased training flags to doc, which was previously missing

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [x] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: .github/workflows/e2e-nvidia-a10g-x1.yml [hunk 1]
```diff
@@ -156,7 +156,7 @@ jobs:
 
           python3.11 -m pip show nvidia-nccl-cu12
           
-          ./scripts/basic-workflow-tests.sh -mf
+          ./scripts/basic-workflow-tests.sh -mL
```
Comment (JamesKunstle) [inline .github/workflows/e2e-nvidia-a10g-x1.yml:159]: Is `e2e-nvidia-a10g-x4` using this flag too?

**Review by JamesKunstle — COMMENTED** [review]
I think this seems solid, I'll run it manually to confirm

File: .github/workflows/e2e-nvidia-a10g-x1.yml (hunk 1 above)
Comment (nathan-weinberg) [reply .github/workflows/e2e-nvidia-a10g-x1.yml:159]: Checking right now

Comment (nathan-weinberg) [reply .github/workflows/e2e-nvidia-a10g-x1.yml:159]: Doesn't look like it

**Review by booxter — APPROVED** [review]

File: docs/ci.md [hunk 2]
```diff
@@ -37,12 +37,13 @@ the E2E job configuration files found
 | Flag | Feature |
 | --- | --- |
 | `e` | Run model evaluation |
-| `m` | Run minimal configuration |
```
[the remark this answers is not in the document]
Comment (nathan-weinberg) [reply docs/ci.md:40]: The script does do that - I added this in https://github.com/instructlab/instructlab/pull/1733 to increase the visibility of the options

**Review by danmcp — APPROVED** [review]

**Pull request merged**